### PR TITLE
Fix regridding test for unstructured nearest regridding on OSX

### PR DESCRIPTION
**Pull request by schlunma**
## Description

This PR adapts coordinates of the input cube for the test to make the regridding consistent across OS.

***

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [ ] [🛠][1] ~All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful~ Only the affected tests are solved

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by valeriupredoi** [comment]
Manu, could you pls temporarily disable or mark xfailed that test that keepd spawning 14 fails just to see all OSC tests pass pls?

**Comment by schlunma** [comment]
Do you want to keep the xfail?

File: .github/workflows/run-tests-monitor.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+      - fix_regrid_tests
```
Comment (valeriupredoi) [inline .github/workflows/run-tests-monitor.yml:7]:
```suggestion
```

**Review by valeriupredoi — APPROVED** [review]
many thanks, Manu! Do we know why the numerical values need be changed in the case of OSX only? Also, about the other tests (the mocker issue ones), it'd be very useful if we understood what the heck's going on there, I'd reckon brushing them under an `xfail` carpet is not quite the right thing to do. I'll have a closer look at the test cases myself, today - black box debugging like I did yesterday just doesn't cut it, it seems

**Comment by schlunma** [comment]
> Do we know why the numerical values need be changed in the case of OSX only?

Yeah, most likely numerical issues. In the original case, one longitude of the source and target grid was 180.0. I guess the float comparison gave different results on Linux than OSX. I changed that now so the edge case where a coordinate of the target grid == coordinate of source grid does not appear anymore.

> Also, about the other tests (the mocker issue ones), it'd be very useful if we understood what the heck's going on there, I'd reckon brushing them under an xfail carpet is not quite the right thing to do. I'll have a closer look at the test cases myself, today - black box debugging like I did yesterday just doesn't cut it, it seems

Yeah, they need to be fixed properly. But do we wanna keep the xfail for now to have green lights?

**Comment by valeriupredoi** [comment]
Yeah sorry, bud let's keep the xfail for now, but we need to make sure we fix those tests sooner than later 🍺
